Build fails if python bindings is turned off

### Describe the bug

As of 3d7a536, build fails with
```
CMake Error at /opt/ttmlir-toolchain/lib/cmake/mlir/AddMLIRPython.cmake:604 (file):
  file failed to create directory:

    /ttir_builder

  because: Permission denied
Call Stack (most recent call first):
  /opt/ttmlir-toolchain/lib/cmake/mlir/AddMLIRPython.cmake:222 (add_mlir_python_sources_target)
  /opt/ttmlir-toolchain/lib/cmake/mlir/AddMLIRPython.cmake:256 (_process_target)
  tools/ttir-builder/CMakeLists.txt:16 (add_mlir_python_modules)
```
if the project was configured with `-DTTMLIR_ENABLE_BINDINGS_PYTHON=OFF`

### Steps to reproduce the issue

_No response_

### Expected behavior

_No response_

### Please complete the following environment information

_No response_

--- a/tools/CMakeLists.txt
+++ b/tools/CMakeLists.txt
@@ -1,12 +1,17 @@
 add_subdirectory(ttmlir-opt)
 add_subdirectory(ttmlir-lsp-server)
-add_subdirectory(ttir-builder)
-add_subdirectory(ttmlir-translate)
-if (TTMLIR_ENABLE_EXPLORER)
-  add_subdirectory(explorer)
-endif()
+if(TTMLIR_ENABLE_BINDINGS_PYTHON AND MLIR_ENABLE_BINDINGS_PYTHON)
+  add_subdirectory(ttir-builder)
 
-if (TTMLIR_ENABLE_PYKERNEL)
-  add_subdirectory(pykernel)
+  if (TTMLIR_ENABLE_PYKERNEL)
+    add_subdirectory(pykernel)
+  endif()
+
+  if (TTMLIR_ENABLE_EXPLORER)
+    add_subdirectory(explorer)
+  endif()
+
 endif()
+add_subdirectory(ttmlir-translate)
+
 add_subdirectory(ttnn-standalone)

